aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -164,6 +164,32 @@ Tensor NestedTensor_gelu(const Tensor& self, c10::string_view approximate) {
       });
 }
 
+Tensor NestedTensor_nested_tensor_from_mask(const Tensor& t, const Tensor& mask, double min_padding_rate) {
+    TORCH_CHECK(mask.scalar_type() == at::ScalarType::Bool, "Mask tensor should be Bool");
+    TORCH_CHECK(mask.dim() == 2, "Padding mask should be 2D");
+    TORCH_CHECK(t.dim() == 3, "Input should be a 3D tensor, N * L * D");
+    // One assumption is mask is always padded at the end , never in the middle
+    auto N = t.size(0), L = t.size(1), D = t.size(2);
+    auto NN = mask.size(0), LL = mask.size(1);
+    TORCH_CHECK(N == NN && L == LL, "Mask size should match input size");
+    // N * L
+    Tensor sizes = mask;
+    sizes.logical_not_();
+    auto real_padding_rate = 1 - (sizes.sum().item().toDouble() / sizes.numel());
+    if (real_padding_rate < min_padding_rate) {
+        return t;
+    }
```

Comment:
As mentioned above: I suggest keeping the padding rate logic outside of this generic op.